compiler/stable_mir/src/compiler_interface.rs

```diff
@@ -56,7 +56,7 @@ pub trait Context {
     fn def_name(&self, def_id: DefId, trimmed: bool) -> Symbol;
 
     /// Returns printable, human readable form of `Span`
-    fn span_to_string(&self, span: Span) -> String;
+    fn span_to_string(&self, span: Span) -> std::borrow::Cow<'static, str>;
```

Comment:
Is this a win? Is there any case where `span_to_string` *doesn't* return a heap allocated `String`?

Reply:
See rustc_span/src/source_map.rs, `no-location` is returned rarely.